lua/ruby_nvim/util.lua

```diff
@@ -101,4 +117,51 @@ M.alternate = function(file_path)
   end
 end
 
+M.all_methods = function ()
+  local bufnr = vim.api.nvim_get_current_buf()
+  local ft = vim.api.nvim_buf_get_option(bufnr, "ft")
+  local tree = vim.treesitter.get_parser(bufnr, ft):parse()[1]
+  local root = tree:root()
+  return get_methods(root)
+end
+
+M.get_gem_name = function (node, bufnr)
+  while node ~= nil do
+    if node:type() == 'call' then
+      local method = node:field('method')[1]
+      local method_name = vim.treesitter.query.get_node_text(method, bufnr)
+      if method_name == 'gem' then
+        local method_args = node:field('arguments')[1] --that when using get_node_text has text 'aws-sdk-s3', '~> 1'
```

Comment:
I loved these comments, they clarify a lot for the reader! Can we work on the wording a little bit?

If the result is a text:

```lua
        local method_args = node:field('arguments')[1] -- e.g. get_node_text returns "'aws-sdk-s3', '~> 1'"
```

Or, if it's a table:

```lua
        local method_args = node:field('arguments')[1] -- e.g. get_node_text returns {'aws-sdk-s3', '~> 1'}
```

If you like it, you can follow this pattern in the next lines (BTW, in line 136 there's a typo `s/usin/using`).